src/psyclone/psyir/nodes/loop.py

```diff
@@ -578,3 +579,53 @@ def gen_code(self, parent):
             if not kernel.module_inline:
                 parent.add(UseGen(parent, name=kernel._module_name, only=True,
                                   funcnames=[kernel._name]))
+
+    @property
+    def is_outermost(self):
+        '''
+        :returns: True if the loop has no ancestors, otherwise False
+        :rtype: bool
+        '''
+        return self.ancestor(Loop) is None
+
+    @property
+    def is_perfectly_nested(self):
```

Comment:
On discussion with @sergisiso I realised I should have asked what the use case for this method is? We've recently improved the 'collapse' option on ParallelLoopTrans so that it is a fair bit smarter:
![image](https://github.com/user-attachments/assets/de827caa-3aa6-4a1c-bb7e-e6377af2a436)
Will that cover what you need or is this new method used in some other way?

Reply:
Oh nice, I wasn't aware of this update - sounds really useful to pass `True`. I'll experiment with this on the UKCA GPU port work and get back to you.

Reply:
I notice that this will insert `collapse(1)` in cases where a collapse isn't possible. I'd thought that was invalid syntax but it seems to compile fine.